nexus/types/src/deployment/planning_input.rs

```diff
@@ -876,20 +893,55 @@ pub struct Policy {
     pub clickhouse_policy: Option<ClickhousePolicy>,
 }
 
-/// Policy for replicated clickhouse setups
 #[derive(Debug, Clone, Serialize, Deserialize)]
 pub struct ClickhousePolicy {
-    /// Should we run the single-node cluster alongside the replicated cluster?
-    /// This is stage 1 of our deployment plan as laid out in RFD 468
-    ///
-    /// If this is set to false, then we will only deploy replicated clusters.
-    pub deploy_with_standalone: bool,
+    pub version: u32,
+    pub mode: ClickhouseMode,
+    pub time_created: DateTime<Utc>,
+}
 
-    /// Desired number of clickhouse servers
-    pub target_servers: usize,
+/// How to deploy clickhouse nodes
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub enum ClickhouseMode {
+    SingleNodeOnly,
+    ClusterOnly { target_servers: u8, target_keepers: u8 },
+    Both { target_servers: u8, target_keepers: u8 },
+}
+
+impl ClickhouseMode {
+    pub fn cluster_enabled(&self) -> bool {
+        match self {
+            ClickhouseMode::SingleNodeOnly => false,
+            _ => true,
```

Comment:
I don't imagine we expect this enum to expand, but I'd probably still spell out the other cases instead of `_ => ...`. (Same comment on line 922 below)

Reply:
Ha, I *knew* you were going to leave this comment.

Reply:
Done in [108ec11](https://github.com/oxidecomputer/omicron/pull/6862/commits/108ec119ea0dee25281e6e1f4031d90ed8afb67a)